vta/hardware/chisel/src/main/scala/Compute.scala

```diff
@@ -0,0 +1,364 @@
+// See LICENSE for license details.
+
+package vta
+
+import chisel3._
+import chisel3.util._
+import freechips.rocketchip.config.Parameters
+
+class ComputeIO(implicit p: Parameters) extends CoreBundle()(p) {
+  val done = new AvalonSlaveIO(dataBits = 1, addrBits = 1)
+  val uops = Flipped(new AvalonSlaveIO(dataBits = 128, addrBits = 32))
+  val biases = Flipped(new AvalonSlaveIO(dataBits = 128, addrBits = 32))
+  val gemm_queue = new AvalonSinkIO(dataBits = 128)
+  val l2g_dep_queue = new AvalonSinkIO(dataBits = 1)
+  val s2g_dep_queue = new AvalonSinkIO(dataBits = 1)
+  val g2l_dep_queue = Flipped(new AvalonSinkIO(dataBits = 1))
+  val g2s_dep_queue = Flipped(new AvalonSinkIO(dataBits = 1))
+  val inp_mem = Flipped(new AvalonSlaveIO(dataBits = 64, addrBits = 15))
+  val wgt_mem = Flipped(new AvalonSlaveIO(dataBits = 64, addrBits = 18))
+  val out_mem = Flipped(new AvalonSlaveIO(dataBits = 128, addrBits = 17))
+}
+
+class DepQueue[T <: Data](gen: T, entries: Int) extends Queue (gen, entries) {}
+class OutQueue[T <: Data](gen: T, entries: Int) extends Queue (gen, entries) {}
+
+class Compute(implicit val p: Parameters) extends Module with CoreParams {
+  val io = IO(new ComputeIO)
+
+  io.inp_mem <> DontCare
+  io.wgt_mem <> DontCare
+
+  val started = !reset.toBool
+
+  val acc_mem = Mem(1 << 8, UInt(512.W))
+  val uop_mem = Mem(1 << 10, UInt(32.W))
+
+  val insn            = Reg(UInt(128.W))
+  val insn_valid      = (insn =/= 0.U) && started
+
+  val opcode          = insn(insn_mem_0_1, insn_mem_0_0)
+  val pop_prev_dep    = insn(insn_mem_1)
+  val pop_next_dep    = insn(insn_mem_2)
+  val push_prev_dep   = insn(insn_mem_3)
+  val push_next_dep   = insn(insn_mem_4)
+
+  val memory_type = insn(insn_mem_5_1, insn_mem_5_0)
+  val sram_base   = insn(insn_mem_6_1, insn_mem_6_0)
+  val dram_base   = insn(insn_mem_7_1, insn_mem_7_0)
+  val y_size      = insn(insn_mem_8_1, insn_mem_8_0)
+  val x_size      = insn(insn_mem_9_1, insn_mem_9_0)
+  val x_stride    = insn(insn_mem_a_1, insn_mem_a_0)
+  val y_pad_0     = insn(insn_mem_b_1, insn_mem_b_0)
+  val y_pad_1     = insn(insn_mem_c_1, insn_mem_c_0)
+  val x_pad_0     = insn(insn_mem_d_1, insn_mem_d_0)
+  val x_pad_1     = insn(insn_mem_e_1, insn_mem_e_0)
```

Comment:
I think that it would be more idiomatic if you represented these a a bundle and removed the `insn_mem_*` variables (which are somewhat confusingly named). For instance,

```scala
class Insn extends Bundle {
	val opcode = UInt(memop_id_bit_width.W)
	val pop_prev_dep = Bool()
    // etc
}

val insn = Reg(UInt(new Insn))
```